src/Aspire.Hosting.MongoDB/MongoDBBuilderExtensions.cs

```diff
@@ -15,22 +17,39 @@ public static class MongoDBBuilderExtensions
     // Internal port is always 27017.
     private const int DefaultContainerPort = 27017;
 
+    private const string UserEnvVarName = "MONGO_INITDB_ROOT_USERNAME";
+    private const string PasswordEnvVarName = "MONGO_INITDB_ROOT_PASSWORD";
+    //MONGO_INITDB_DATABASE ???
+
     /// <summary>
     /// Adds a MongoDB resource to the application model. A container is used for local development. This version the package defaults to the 7.0.8 tag of the mongo container image.
     /// </summary>
     /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
     /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
     /// <param name="port">The host port for MongoDB.</param>
+    /// <param name="userName">A parameter that contains the MongoDb server user name, or <see langword="null"/> to use a default value.</param>
+    /// <param name="password">A parameter that contains the MongoDb server password, or <see langword="null"/> to use a generated password.</param>
     /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
-    public static IResourceBuilder<MongoDBServerResource> AddMongoDB(this IDistributedApplicationBuilder builder, string name, int? port = null)
+    public static IResourceBuilder<MongoDBServerResource> AddMongoDB(this IDistributedApplicationBuilder builder,
```

Comment:
This is also a binary breaking change.

Reply:
Is it better to add a new overload with new parameters?

Reply:
I'm not sure if we can avoid the breaking change here. If we add another overload with optional arguments how would you disambiguate between it and the new overload with the optional password arguments?

Reply:
I agree; otherwise, maybe the best option will be to get back to the original implementation and add an extension method of IResourceBuilder<MongoDBServerResource> (like "WithAuthentication(user?, pwd?)" ) to enable authentication for the MongoDB container. With this approach, we don't have any break changes, and we activate authentication only when the method is called (even if I wonder if running db without authentication is part of good practice:) ). Do you think this approach is more aligned?

Reply:
I think we should switch mongodb to always have a password (runtime breaking change in the name of security). The key here is to avoid removing the original method that will result in a method missing exception for those that compiled against the old one. To add a new method, we need the old method to make all of the other arguments non-optional and a new method with all of the parameters.

```C#
public static IResourceBuilder< MongoDBServerResource> AddMongoDb(this IDistributedApplicationBuilder builder, string name, int? port);
public static IResourceBuilder< MongoDBServerResource> AddMongoDb(this IDistributedApplicationBuilder builder, string name, int? port = null, IResourceBuilder<ParameterResource>? userName = null, IResourceBuilder<ParameterResource>? password = null);
```

Reply:
@davidfowl I'm okay with switching MongoBd secure by default, but in this case, I can't be fully secure without breaking change on MongoDBServerResource. Otherwise, someone can use the primary constructor with name only and default to MongoDBServerResource without username and password, even if I think nobody tries to create a MongoDBServerResource instance by hand. Let me know if you prefer I secure MongoDBServerResource also or if I code the version without breaking changes.

Reply:
If you manually create the resource, yes you're right. We can obsolete the constructor with no arguments and remove it in a later release. These 2 methods will always set the password.